packages/ariakit/src/focus-trap/focus-trap-region.tsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import { getFirstTabbableIn, getLastTabbableIn } from "ariakit-utils/focus";
+import { useForkRef, useWrapElement } from "ariakit-utils/hooks";
+import {
+  createComponent,
+  createElement,
+  createHook,
+} from "ariakit-utils/system";
+import { As, Options, Props } from "ariakit-utils/types";
+import { FocusTrap } from "ariakit/focus-trap";
+
+/**
+ * A component hook that returns props that can be passed to `Role` or any other
+ * Ariakit component to render a focus trap region element.
+ * @see https://ariakit.org/components/focus-trap-region
+ * @example
+ * ```jsx
+ * const props = useFocusTrapRegion();
+ * <Role {...props} />
+ * ```
+ */
+export const useFocusTrapRegion = createHook<FocusTrapRegionOptions>(
+  ({ enabled = false, ...props }) => {
+    const container = React.useRef<HTMLDivElement>();
+
+    props = useWrapElement(
+      props,
+      (element) => {
+        const renderFocusTrap = () => {
+          if (!enabled) return null;
+          return (
+            <FocusTrap
+              onFocus={(event) => {
+                if (!container.current) return;
+                const first = getFirstTabbableIn(container.current, true);
+                const last = getLastTabbableIn(container.current, true);
```

Comment:
In this case, I guess your idea of calling `getAllTabbableIn` once would make more sense.

Reply:
Right!